### PR TITLE
go mod tidy + ci check for site build

### PR DESCRIPTION
**Pull request by eddie-knight**

### CODE REVIEW

File: .github/workflows/test-build.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+permissions:
+  contents: read
+
+jobs:
+  # Build job
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```
Comment (funnelfiasco) [inline .github/workflows/test-build.yml:16]: Let's have this action drop the credentials after it's done by adding

```
        with:
            persist-credentials: false
```

See https://github.com/ossf/security-baseline/blob/main/.github/workflows/web-publish.yml#L27-L30

**Review by funnelfiasco — APPROVED** [review]
LGTM. This is a good addition. I think you should make the suggested change re: persistent credentials, but I won't block on it.

It would be good as a next step to have some way to view the rendered output so that we can find more subtle issues, but this will at least catch the glaring ones.

File: .github/workflows/test-build.yml (hunk 1 above)
Comment (eddie-knight) [reply .github/workflows/test-build.yml:16]:
```suggestion
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
        with:
            persist-credentials: false
```

Like so?

Comment (funnelfiasco) [reply .github/workflows/test-build.yml:16]: #nailedit

**Review by SecurityCRob — APPROVED** [review]

**Pull request merged**